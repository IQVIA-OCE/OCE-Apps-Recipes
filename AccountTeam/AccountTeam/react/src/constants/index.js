import { Platform } from 'react-native';

export { SORT_OPTION_LIST, ACCESS_LEVEL_PICKLIST } from './sortOptionsList';
export { NAMESPACE, LOCALIZATION_NAMESPACE } from './namespacePrefix';
export { LOADING_STATUS } from './loadingStatus';
export const isIphone = !Platform.isPad && Platform.OS !== 'web';
