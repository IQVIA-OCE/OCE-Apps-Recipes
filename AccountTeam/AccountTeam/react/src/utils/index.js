export { default as api } from './api';
export { isIphone } from './common';
export { splitStringFromArray, splitStringByDelimiter } from './helper'