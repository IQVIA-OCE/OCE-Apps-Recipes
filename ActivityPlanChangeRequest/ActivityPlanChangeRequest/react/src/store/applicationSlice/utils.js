export const getPicklistValues = (described) => {
  const obj = {};

  described.fields.forEach((field) => {
    if (field.picklistValues) {
      obj[field.name] = field.picklistValues.map(
        (valueObject) => valueObject.value
      );
    }
  });

  return obj;
};

export const getAccountsValues = (list) => {
  return list.map((account) => ({
    label: account.Name,
    value: account.Id,
  }));
};
