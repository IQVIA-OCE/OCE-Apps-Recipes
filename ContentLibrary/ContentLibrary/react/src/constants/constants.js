export const OWNED_BY_ME = 'ownedByMe';
export const SHARED_WITH_ME = 'sharedWithMe';
export const NO_CONNECTION = 'No Connection';