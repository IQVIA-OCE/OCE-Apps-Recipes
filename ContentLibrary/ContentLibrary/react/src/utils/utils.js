
const normalizeAttachmentsByOrders = (response, localAttachments = []) => {
    const data = [];
    response.map(item => {
        if (!item.done) {
            item.map(element => {
                data.push({
                    ContentDocument: {
                        ...element.ContentDocument,
                        synced: localAttachments.find(el => el.Id === element.Id)
                    },
                    Id: element.Id,
                });
            });
        }
    })
    return data;
};

const normalizeSearchAttachmentOwnedByMe = (response, attachmentsList) => {
    return response[0].map(item => ({
        ...item,
        Added: attachmentsList.some(el => el.ContentDocument.Id === item.Id),
    }));
}

const normalizeSearchAttachmentSharedWithMe = (response, attachmentsList) => {
    return response[0].map(item => {
        return {
            FileExtension: item.ContentDocument.FileType,
            Id: item.ContentDocument.Id,
            Title: item.ContentDocument.Title,
            Added: attachmentsList.some(el => el.ContentDocument.Id === item.ContentDocument.Id)
        }
    });
}

export { normalizeAttachmentsByOrders, normalizeSearchAttachmentOwnedByMe, normalizeSearchAttachmentSharedWithMe };