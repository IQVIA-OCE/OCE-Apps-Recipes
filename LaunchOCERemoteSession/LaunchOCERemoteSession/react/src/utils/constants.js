export const ListItemType = {
    Meeting: 'Meeting',
    Call: 'Call'
}