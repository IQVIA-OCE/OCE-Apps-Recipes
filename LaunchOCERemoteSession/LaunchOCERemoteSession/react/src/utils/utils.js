import { DateTime } from "luxon";
import { environment } from "oce-apps-bridges";
import { ListItemType } from "./constants";
import { NAMESPACE } from "../../App";

export const normalizeCalls = (callsData) =>
  callsData.map((el) => ({
    id: el.Id,
    name: el.Name,
    duration: el[`${NAMESPACE}Duration__c`],
    accountId: el[`${NAMESPACE}Account__c`],
    dateTime: DateTime.fromISO(el[`${NAMESPACE}CallDateTime__c`], {
      zone: environment.timeZone(),
    }).toFormat("hh:mm a"),
    endDateTime: DateTime.fromISO(el[`${NAMESPACE}EndCallDateTime__c`], {
      zone: environment.timeZone(),
    }).toFormat("hh:mm a"),
    accountName: el[`${NAMESPACE}Account__r.Name`] || el[`${NAMESPACE}Account__r`].Name,
    location: el[`${NAMESPACE}LocationName__c`],
    endDateTimeFull: DateTime.fromISO(el[`${NAMESPACE}EndCallDateTime__c`], {
      zone: environment.timeZone(),
    }),
    startDateTimeFull: DateTime.fromISO(el[`${NAMESPACE}CallDateTime__c`], {
      zone: environment.timeZone(),
    }),
    signature: el[`${NAMESPACE}Signature__c`],
    type: ListItemType.Call,
  }));

export const normalizeMeetings = (meetingsData) =>
  meetingsData.map((el) => ({
    id: el.Id,
    name: el.Name,
    dateTime: DateTime.fromISO(el[`${NAMESPACE}StartDateTime__c`], {
      zone: environment.timeZone(),
    }).toFormat("hh:mm a"),
    endDateTime: DateTime.fromISO(el[`${NAMESPACE}EndDateTime__c`], {
      zone: environment.timeZone(),
    }).toFormat("hh:mm a"),
    accountName: el.Name,
    location: el[`${NAMESPACE}Location__c`],
    endDateTimeFull: DateTime.fromISO(el[`${NAMESPACE}EndDateTime__c`], {
      zone: environment.timeZone(),
    }),
    startDateTimeFull: DateTime.fromISO(el[`${NAMESPACE}StartDateTime__c`], {
      zone: environment.timeZone(),
    }),
    type: ListItemType.Meeting,
  }));
