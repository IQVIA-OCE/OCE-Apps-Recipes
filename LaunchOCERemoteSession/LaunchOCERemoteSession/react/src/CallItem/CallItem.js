import React from 'react';
import { View, StyleSheet, Text, TouchableOpacity } from 'react-native';
import { IconButton } from 'apollo-react-native';
import { NavigationActions } from '../../bridge/Navigation/NavigationActions';
import { navigator } from '../../bridge/Navigation/ScreenNavigator';
import { localized } from '../../bridge/Localization/localization.native';
import { NAMESPACE } from '../../App';
import { DateTime } from 'luxon';
import { ListItemType } from './../utils/constants';

const CallItem = ({ item, setErrorBanner, filterItemsById }) => {
  const {
    id,
    dateTime,
    endDateTime,
    accountName,
    location,
    endDateTimeFull,
    name,
    type,
    signature,
  } = item;

  const showItem = (id) => {
    const entity =
      type == ListItemType.Call
        ? `${NAMESPACE}Call__c`
        : `${NAMESPACE}Meeting__c`;

    const actions = NavigationActions.details({
      entity,
      id: id,
    });

    navigator.dispatch(actions).then(
      (r) => console.log(r),
      (e) =>
        setErrorBanner({
          isVisible: true,
          message: e,
        })
    );
  };

  const launchCall = (id, endDateTimeFull) => {
    const endTime = DateTime.fromISO(endDateTimeFull).toMillis();
    const currentTime = DateTime.local().toMillis();
    const expiredLinkText = {
      message: localized(`${NAMESPACE}linkexpired`, 'The link is expired'),
    };

    if (currentTime > endTime) {
      setErrorBanner({
        isVisible: true,
        message: expiredLinkText,
      });
      filterItemsById(id);
    } else {
      const callUrl = `oce://deeplink/modal/sobject-name/${NAMESPACE}Call__c/record-id/${id}/action/view?commands=launchcall`;
      const meetingUrl = `oce://deeplink/modal/sobject-name/${NAMESPACE}Meeting__c/record-id/${id}/action/view?commands=launchmeeting`;
      const url = type == ListItemType.Call ? callUrl : meetingUrl;
      navigator.openDeeplink(url).then(
        (r) => console.log(r),
        (e) =>
          setErrorBanner({
            isVisible: true,
            message: e,
          })
      );
    }
  };

  const isItemSyncedWithSf = id
    ? !id.includes('meeting__c') && !id.includes('call__c')
    : false;

  return (
    <TouchableOpacity
      onPress={() => showItem(id)}
      disabled={!isItemSyncedWithSf || (signature && signature.length > 0)}
    >
      <View style={styles.row}>
        <View
          style={{
            width: '70%',
            alignContent: 'center',
            justifyContent: 'center',
          }}
        >
          <View>
            <Text style={{ fontWeight: '500', fontSize: 15 }}>
              {accountName}
            </Text>
            <Text style={styles.dateTime}>
              {dateTime} - {endDateTime}
            </Text>
            {location ? (
              <Text style={{ fontWeight: '500', fontSize: 12 }}>
                {location}
              </Text>
            ) : null}
            {!isItemSyncedWithSf ? (
              <Text style={{ fontWeight: '300', fontSize: 12 }}>
                {localized('sync_required', 'Synchronization Required')}
              </Text>
            ) : null}
            {signature && signature.length ? (
              <Text style={{ fontWeight: '300', fontSize: 12 }}>
                Signed Call
              </Text>
            ) : null}
          </View>
        </View>
        <View style={{ alignItems: 'center' }}>
          <IconButton
            icon={
              type == ListItemType.Call ? 'cellphone-wireless' : 'microphone'
            }
            size={30}
            style={{ width: 35, height: 40 }}
            onPress={() => {
              launchCall(id, endDateTimeFull);
            }}
            disabled={!isItemSyncedWithSf || (signature && signature.length > 0)}
          />
          <Text style={styles.launchText}>
            {type == ListItemType.Call
              ? localized(`${NAMESPACE}launchcall`, 'Launch Call')
              : localized(
                  `${NAMESPACE}launch_remote_meeting`,
                  'Launch Remote Meeting'
                )}
          </Text>
        </View>
      </View>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  row: {
    marginVertical: 5,
    marginHorizontal: 10,
    padding: 10,
    borderBottomColor: '#dcdcdc',
    borderBottomWidth: 1,
    justifyContent: 'space-between',
    flexDirection: 'row',
  },
  dateTime: {
    fontSize: 13,
    color: '#666',
  },
  launchText: {
    fontSize: 10,
    width: 100,
    textAlign: 'center',
  },
});

export default CallItem;
