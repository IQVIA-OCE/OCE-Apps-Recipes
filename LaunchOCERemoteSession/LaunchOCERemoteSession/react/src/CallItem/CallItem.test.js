import React from "react";
import renderer, { act } from "react-test-renderer";
import CallItem from "./CallItem";
import { TouchableOpacity } from "react-native";
import { IconButton } from "apollo-react-native";
import { navigator } from "oce-apps-bridges";
import { ListItemType } from "../utils/constants";


describe("CallItem", () => {
  beforeAll(() => {
    jest.useFakeTimers();
    navigator.dispatch = jest.fn().mockResolvedValueOnce("");
    navigator.openDeeplink = jest.fn().mockResolvedValueOnce("");
  });
  it("should render component of call type", async () => {
    const promise = Promise.resolve();
    let tree;
    act(() => {
      tree = renderer.create(
        <CallItem
          item={{
            id: "id",
            dateTime: "",
            endDateTime: "",
            accountName: "accountName",
            location: "location",
            type: ListItemType.Call
          }}
          setErrorBanner={jest.fn()}
        />
      );
    });

    act(() => tree.root.findAllByType(TouchableOpacity)[0].props.onPress());
    act(() => tree.root.findAllByType(IconButton)[0].props.onPress());

    navigator.dispatch = jest
      .fn()
      .mockRejectedValueOnce(new Error("Error message"));
    navigator.openDeeplink = jest
      .fn()
      .mockRejectedValueOnce(new Error("Error message"));

    act(() => tree.root.findAllByType(TouchableOpacity)[0].props.onPress());
    act(() => tree.root.findAllByType(IconButton)[0].props.onPress());

    await act(() => promise);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("should render component of meeting type", async () => {
    let tree;
    act(() => {
      tree = renderer.create(
        <CallItem
          item={{
            id: "id",
            dateTime: "",
            endDateTime: "",
            accountName: "accountName",
            location: "location",
            type: ListItemType.Meeting
          }}
          setErrorBanner={jest.fn()}
        />
      );
    });

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
