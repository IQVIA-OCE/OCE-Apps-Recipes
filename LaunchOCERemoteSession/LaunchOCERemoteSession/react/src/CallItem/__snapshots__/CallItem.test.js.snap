// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CallItem should render component of call type 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#dcdcdc",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": 10,
        "marginVertical": 5,
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignContent": "center",
          "justifyContent": "center",
          "width": "70%",
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "fontSize": 15,
              "fontWeight": "500",
            }
          }
        >
          accountName
        </Text>
        <Text
          style={
            Object {
              "color": "#666",
              "fontSize": 13,
            }
          }
        >
          
           - 
          
        </Text>
        <Text
          style={
            Object {
              "fontSize": 12,
              "fontWeight": "500",
            }
          }
        >
          location
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        centered={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        nativeID="animatedComponent"
        onClick={[Function]}
        onLayout={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        pointerEvents="box-only"
        style={Object {}}
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 22.5,
                  "height": 45,
                  "width": 45,
                },
                undefined,
                Object {
                  "height": 40,
                  "width": 35,
                },
              ],
            ]
          }
        >
          <View>
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontSize": 30,
                  },
                  Array [
                    Object {
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "bottom": 0,
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "borderRadius": 0,
              },
            ]
          }
        />
      </View>
      <Text
        style={
          Object {
            "fontSize": 10,
            "textAlign": "center",
            "width": 100,
          }
        }
      >
        Launch Call
      </Text>
    </View>
  </View>
</View>
`;

exports[`CallItem should render component of meeting type 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#dcdcdc",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": 10,
        "marginVertical": 5,
        "padding": 10,
      }
    }
  >
    <View
      style={
        Object {
          "alignContent": "center",
          "justifyContent": "center",
          "width": "70%",
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "fontSize": 15,
              "fontWeight": "500",
            }
          }
        >
          accountName
        </Text>
        <Text
          style={
            Object {
              "color": "#666",
              "fontSize": 13,
            }
          }
        >
          
           - 
          
        </Text>
        <Text
          style={
            Object {
              "fontSize": 12,
              "fontWeight": "500",
            }
          }
        >
          location
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": false,
          }
        }
        accessible={true}
        centered={true}
        collapsable={false}
        focusable={true}
        hitSlop={
          Object {
            "bottom": 6,
            "left": 6,
            "right": 6,
            "top": 6,
          }
        }
        nativeID="animatedComponent"
        onClick={[Function]}
        onLayout={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        pointerEvents="box-only"
        style={Object {}}
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "margin": 6,
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 22.5,
                  "height": 45,
                  "width": 45,
                },
                undefined,
                Object {
                  "height": 40,
                  "width": 35,
                },
              ],
            ]
          }
        >
          <View>
            <Text
              accessibilityElementsHidden={true}
              allowFontScaling={false}
              importantForAccessibility="no-hide-descendants"
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "color": "#000000",
                    "fontSize": 30,
                  },
                  Array [
                    Object {
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                    },
                    Object {
                      "backgroundColor": "transparent",
                    },
                  ],
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "bottom": 0,
                "left": 0,
                "overflow": "hidden",
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              Object {
                "borderRadius": 0,
              },
            ]
          }
        />
      </View>
      <Text
        style={
          Object {
            "fontSize": 10,
            "textAlign": "center",
            "width": 100,
          }
        }
      >
        Launch Remote Meeting
      </Text>
    </View>
  </View>
</View>
`;
